### PR TITLE
task fruits finished

### PR DESCRIPTION
**Pull request by Taniann**

### CODE REVIEW

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Fruit.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    public LocalDate getDeliveryDate() {
+        return deliveryDate;
+    }
+
+    public BigDecimal getPrice() {
+        return price;
+    }
+
+    public void setPrice(BigDecimal price) {
```
Comment (un-install) [inline src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Fruit.java:38]: You should store only immutable objects in collections. I think setters are not allowed

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+     * method makes sale for transfered kinds of fruits,
+     * if expiration date less than 7 days
+     * the price of fruit will be reduced by 25 percent
+     * @param kinds
+    */
+    public void makeSale(List<FruitKind> kinds) {
+        LocalDate now = LocalDate.now();
+        fruits.stream().
+                filter(fruit -> kinds.contains(fruit.getKind()) && fruit.getExpirationDate() - DAYS.between(fruit.getDeliveryDate(), now) <= DAYS_FOR_SALE).
+                forEach(fruit -> fruit.setPrice(fruit.getPrice().multiply(MULTIPLIER_FOR_SALE)));
```
Comment (un-install) [inline src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java]: you better create new object (smth like fruit -> new Fruit(...)) and translate fields from fruit to constructor via getters except price

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    public  List<Fruit> getFruits() {
+        return fruits;
+    }
+
+    public void addFruit(Fruit fruit) {
+        if (!fruits.contains(fruit))
+            fruits.add(fruit);
+    }
+
+    public List<Fruit> getFreshFruits(LocalDate date) {
```
Comment (un-install) [inline src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java:27]: Why you need two methods wich doing the same?

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java (hunk 2 above)
Comment (un-install) [inline src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java]: You should recheck the task about 5th method. There are required another set of parameters

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/FruitKind.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ */
+public enum FruitKind{
+    APPLE("grenni"),
+    PEAR("bera"),
+    ORANGE(""),
+    BANANA(""),
+    GRAPES("kysh-mysh");
+
+    FruitKind(String name) {
+        this.name = name;
```
Comment (un-install) [inline src/main/java/ua/mateacademy/tania/lesson5/taskfruits/FruitKind.java:14]: You newer use the fields of enum. I think task was not only create them

**Review by un-install — CHANGES_REQUESTED** [review]
The main reason you see this message is setter in better-to-be-immutble class

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java (hunk 2 above)
Comment (un-install) [reply src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java]: and why you dont use varagrs (For example Object... args)

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Fruit.java (hunk 1 above)
Comment (Taniann) [reply src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Fruit.java:38]: I understand you, but I need change price for making sale. And I guess we shod`nt use muttable class in key of HashMap and in HashSet. But in List in some case we can do it.

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java (hunk 2 above)
Comment (Taniann) [reply src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java]: I didnt create a new object here. I check if fruict has expiration date less than 7 days, then price will be changed.

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java (hunk 3 above)
Comment (Taniann) [reply src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java:27]: it`s true. first method without lambda, second with lambda. I did it to check myself :)

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java (hunk 2 above)
Comment (Taniann) [reply src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java]: The task was to do it with some kind of fruits, and I did it for some kind of fruits

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/FruitKind.java (hunk 4 above)
Comment (Taniann) [reply src/main/java/ua/mateacademy/tania/lesson5/taskfruits/FruitKind.java:14]: it`s true, I never use field, I will change it

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Fruit.java (hunk 1 above)
Comment (SerhiiPasko) [reply src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Fruit.java:38]: Tania is true

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/FruitKind.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package ua.mateacademy.tania.lesson5.taskfruits;
+
+/**
+ * Created by Tania Nebesna on 04.01.2019.
+ */
+public enum FruitKind{
+    APPLE("grenni"),
+    PEAR("bera"),
+    ORANGE(""),
```
Comment (SerhiiPasko) [inline src/main/java/ua/mateacademy/tania/lesson5/taskfruits/FruitKind.java]: Newer do that - ""

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java (hunk 2 above)
Comment (un-install) [reply src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java]: 5. Якщо срок реалізації менше певного часу (параметр) то змінити ціну (параметр процент), але це стосується тільки певних видів фруктів
As I understand "параметр" means method argument. isnt it so?

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    private static final int DAYS_FOR_SALE = 7;
+    private static final BigDecimal MULTIPLIER_FOR_SALE = BigDecimal.valueOf(0.75);
+
+    private List<Fruit> fruits = new ArrayList<>();
+
+    public  List<Fruit> getFruits() {
+        return fruits;
+    }
+
+    public void addFruit(Fruit fruit) {
```
Comment (SerhiiPasko) [inline src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java:23]: You would like use List as Set. It incorrectly, change

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+                collect(Collectors.toList());
+    }
+
+    /**
+     * method makes sale for transfered kinds of fruits,
+     * if expiration date less than 7 days
+     * the price of fruit will be reduced by 25 percent
+     * @param kinds
+    */
+    public void makeSale(List<FruitKind> kinds) {
```
Comment (SerhiiPasko) [inline src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java]: Your method change your input values. It is incorrect. What will be if you use this method twice? Return list of results (new List)

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Read comments

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Fruit.java (hunk 1 above)
Comment (un-install) [reply src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Fruit.java:38]: Not only set. Many methods of List also use hashcode and equals, and acording to this, there can be some very unexpectedly bugs. 
Anyway solution is right below)

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java (hunk 2 above)
Comment (un-install) [reply src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Shop.java]: ^solution

File: src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Fruit.java (hunk 1 above)
Comment (un-install) [reply src/main/java/ua/mateacademy/tania/lesson5/taskfruits/Fruit.java:38]: Ok

**Review by SerhiiPasko — APPROVED** [review]
Good

**Review by un-install — APPROVED** [review]

**Pull request merged**